cli/deployment/utils.go

```diff
@@ -25,13 +30,87 @@ func unmarshal(data string) (map[string]interface{}, error) {
 // after import "absolutisation" deployment config might be copied to any location (as current CFT cli will copy its
 // copy to tmp folder.
 // Examples:
-//    /base/folder/config.YAML and ../script.py will concatenate to /base/script.py
-//    /base/folder/config.YAML and /base/folder/script.py will concatenate to /base/folder/script.py as long path already absolute
-func ReparentPath(parent string, child string) string {
-	// check if file already has absolute path
-	if child[0] == os.PathSeparator {
-		return child
+//    /base/folder/config.yaml and ../script.py will concatenate to /base/script.py
+//    /base/folder/config.yaml and /base/folder/script.py will concatenate to /base/folder/script.py as long path already absolute
+func ReparentPath(baseDir string, file string) string {
+	// check if import statement path already absolute
+	if file[0] == os.PathSeparator {
+		return file
 	}
-	dir := filepath.Dir(parent)
-	return filepath.Clean(filepath.Join(dir, child))
+	baseDir, _ = filepath.Abs(baseDir)
+	baseDir = filepath.Dir(baseDir)
+	relative := filepath.Clean(filepath.Join(baseDir, file))
+	result, err := filepath.Abs(relative)
+	if err != nil {
+		log.Fatalf("error creating absolute path, for file: %s, error: %v", relative, err)
+	}
+	return result
+}
+
+// check if string is yaml
+func IsYAML(text string) bool {
+	obj := struct{}{}
+	err := yaml.Unmarshal([]byte(text), obj)
+	return err == nil
+}
+
+// append string map B to map A, returns A
+func AppendMap(a map[string]string, b map[string]string) map[string]string {
+	for k, v := range b {
+		a[k] = v
+	}
+	return a
+}
+
+/*
+creates valid deployment name from file path satisfied Deployment resource "name" field requirements:
+Specifically, the name must be 1-63 characters long and match the regular expression [a-z]([-a-z0-9]*[a-z0-9])?
+which means the first character must be a lowercase letter,
+and all following characters must be a dash, lowercase letter, or digit, except the last character, which cannot be a dash.
+see more https://cloud.google.com/deployment-manager/docs/reference/latest/deployments#resource
+*/
```

Comment:
Please use line comments as suggested previously and follow doc comment convention as described at https://golang.org/doc/effective_go.html#commentary

Reply:
Also note in the documentation that this function truncates the name.